### PR TITLE
Sprint 6 sale

### PR DESCRIPTION
**Pull request by AbdulSir**

### CODE REVIEW

**Comment by TZnoj** [comment]
#53 I went over the code and noticed that edited save values don't automatically refresh as they should. I created a bug report for the issue.

File: backend/controllers/saleController.js
Comment (purplesmurf1998) [inline backend/controllers/saleController.js:54]: Instead of populating manually the different properties, you could take advantage of the `advancedFilter` util function.

File: backend/controllers/saleController.js
Comment (purplesmurf1998) [inline backend/controllers/saleController.js:74]: Same comment as for lines 51 - 54

File: frontend/src/views/pages/inventory/InventorySlot.vue
Comment (purplesmurf1998) [inline frontend/src/views/pages/inventory/InventorySlot.vue:2]: SonarCloud flagged this line as a security hotspot because you are using the `v-html` prop. A way to solve this would be by putting the html value directly in the `<td></td>` tags like this: `<td>{{ value }}</td>`

**Review by purplesmurf1998 — CHANGES_REQUESTED** [review]
I also noticed when a vehicle is delivered, we can still modify it's information and request sales. Any edit buttons on the vehicle details page should be invisible once the vehicle has been delivered.

File: backend/controllers/saleController.js
Comment (AbdulSir) [reply backend/controllers/saleController.js:54]: I'm not sure how to leverage the advancedFilter fucntion. Could you provide an example?

Comment (purplesmurf1998) [reply backend/controllers/saleController.js:54]: You could use something like this 
`const query = await advancedFilter(Sale, req.query)`

However, adding this change means you need to modify your API request (axios) by adding the populate argument as a query parameter like so:
`GET /api/v1/inventory/details/sale/:dealershipId/?populate=vehicle,sales_rep,approved_by`.

This makes the controller more flexible in the future. If not, the lines of code could also be reduced by simply using `.populate('vehicle sales_rep approved_by')`

Comment (AbdulSir) [reply backend/controllers/saleController.js]: Thanks for the example. I have opted for the second option (i.e. populate separated by space). This is address in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/52/commits/6d067bd2b87c4a5d34eede74a10719c101e6f202)

File: frontend/src/views/pages/inventory/InventorySlot.vue
Comment (AbdulSir) [reply frontend/src/views/pages/inventory/InventorySlot.vue]: This is address in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/52/commits/6d067bd2b87c4a5d34eede74a10719c101e6f202)

**Review by purplesmurf1998 — CHANGES_REQUESTED** [review]
I've found a bug when testing the PR, you can find it [here](https://github.com/purplesmurf1998/Autotracks/issues/54). Will approve once this issue is addressed.

File: frontend/src/views/pages/transactions/TransactionDetails.vue
Comment (purplesmurf1998) [inline frontend/src/views/pages/transactions/TransactionDetails.vue:303]: Instead of setting the sales table as a new object from the response, you can simply call the fetchSales() method that is passed down from the TransactionTable parent component. This way the table refreshes with the proper formatting.

**Review by purplesmurf1998 — CHANGES_REQUESTED** [review]
# General comments about the implementation:

1. Managers should be required when creating a sale transaction because every sale needs to be approved by a manager.
2. The option to mark a vehicle as delivered should also be in the actions dropdown inside the vehicle details page once the sale has been approved

File: frontend/src/views/pages/transactions/TransactionDetails.vue
Comment (AbdulSir) [reply frontend/src/views/pages/transactions/TransactionDetails.vue]: This is a good idea. The code has been updated to follow this suggestion in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/52/commits/3e369c5152623fbe180c6911d67c2c09cb5b502f)

**Review by purplesmurf1998 — APPROVED** [review]
The changes and bug fixes look good. I've tested again on my end and everything runs smoothely.

**Comment by TZnoj** [comment]
As all the bugs have resolved, this branch is ready to be merged. I have checked all the functionality and it meets the acceptance requirements.

**Review by TZnoj — APPROVED** [review]

**Pull request merged**
